libusb/core.c

```diff
@@ -1006,6 +1007,29 @@ uint8_t API_EXPORTED libusb_get_device_address(libusb_device *dev)
 	return dev->device_address;
 }
 
+/** \ingroup libusb_dev
+ * Get a platform id string of the device on the bus it is connected to.
+ *
+ * Linux:   udev_device_new_from_subsystem_sysname(,"usb",id);
+ * macOS:   IORegistryEntryIDMatching(id);
```

Comment:
Code doesn't seem to use IORegistryEntryIDMatching, but rather uses IORegistryEntryGetRegistryEntryID.

Also, since macOS is the odd man out here, where the native API uses uint64_t instead of a string, I think some elaboration would be nice. A note to say that the string is an ASCII decimal representation of the uint64_t returned by IORegistryEntryGetRegistryEntryID.

Reply:
Maybe worth adding the string to int conversation to this snippet, users will probably just copy what is here, I guess strtoull?

I like the note, but I would put it with the macos bit (imagine there are many more platforms added in future)

Reply:
I can use `IORegistryEntryIDMatching(strtoull(path, NULL, 10))`, but `strtoull` should still be error-checked. Is that ok?